spec.html

```diff
@@ -42733,6 +42733,40 @@ <h1>
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-arraybuffercopyanddetach" type="abstract operation">
+        <h1>
+          ArrayBufferCopyAndDetach (
+            _arrayBuffer_: an ECMAScript language value,
+            _newLength_: an ECMAScript language value,
+            _preserveResizability_: ~preserve-resizability~ or ~fixed-length~,
+          ): either a normal completion containing an ArrayBuffer or a throw completion
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. Perform ? RequireInternalSlot(_arrayBuffer_, [[ArrayBufferData]]).
+          1. If IsSharedArrayBuffer(_arrayBuffer_) is *true*, throw a *TypeError* exception.
+          1. If _newLength_ is *undefined*, then
+            1. Let _newByteLength_ be _arrayBuffer_.[[ArrayBufferByteLength]].
+          1. Else,
+            1. Let _newByteLength_ be ? ToIndex(_newLength_).
```

Comment:
I'd personally pull this parameter validation/coercing out so this AO can have narrower parameter types. But I understand it's desirable to deduplicate it and not worth introducing another AO for that.

Side note: it's unfortunate we're coercing to numbers indiscriminately in new APIs. I can't wait until we've agreed to stop doing this.

Reply:
I don't mind making the change if folks prefer it; it can also be done in a follow-on.

Reply:
I wouldn't bother refactoring I think. It's unfortunate but none of the alternatives are better.